collect_app/src/main/java/org/odk/collect/android/activities/viewmodels/FormDownloadListViewModel.java

```diff
@@ -0,0 +1,210 @@
+ */
+
+package org.odk.collect.android.activities.viewmodels;
+
+import android.arch.lifecycle.ViewModel;
+
+import org.odk.collect.android.R;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.logic.FormDetails;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+
+public class FormDownloadListViewModel extends ViewModel {
+    private HashMap<String, FormDetails> formNamesAndURLs = new HashMap<>();
+    private final HashMap<String, Boolean> formResult = new HashMap<>();
+
+    private final ArrayList<HashMap<String, String>> formList = new ArrayList<>();
+
+    private final LinkedHashSet<String> selectedForms = new LinkedHashSet<>();
+
+    private String[] formIdsToDownload;
+
+    private String alertTitle;
+    private String progressDialogMsg;
+    private String alertDialogMsg;
+    private String url;
+    private String username;
+    private String password;
+
+    private boolean progressDialogShowing;
+    private boolean alertShowing;
+    private boolean cancelDialogShowing;
+    private boolean shouldExit;
+    private boolean loadingCanceled;
+    // Variables for the external app intent call
+    private boolean isDownloadOnlyMode;
+
+    public HashMap<String, FormDetails> getFormNamesAndURLs() {
+        return formNamesAndURLs;
+    }
+
+    public void setFormNamesAndURLs(HashMap<String, FormDetails> formNamesAndURLs) {
+        this.formNamesAndURLs = formNamesAndURLs;
+    }
+
+    public void clearFormNamesAndURLs() {
+        formNamesAndURLs.clear();
+    }
+
+    public String getAlertTitle() {
+        return alertTitle;
+    }
+
+    public void setAlertTitle(String alertTitle) {
+        this.alertTitle = alertTitle;
+    }
+
+    public String getProgressDialogMsg() {
+        return progressDialogMsg == null ? Collect.getInstance().getString(R.string.please_wait) : progressDialogMsg;
+    }
+
+    public void setProgressDialogMsg(String progressDialogMsg) {
+        this.progressDialogMsg = progressDialogMsg;
+    }
+
+    public String getAlertDialogMsg() {
+        return alertDialogMsg;
+    }
+
+    public void setAlertDialogMsg(String alertDialogMsg) {
+        this.alertDialogMsg = alertDialogMsg;
+    }
+
+    public boolean isAlertShowing() {
+        return alertShowing;
+    }
+
+    public void setAlertShowing(boolean alertShowing) {
+        this.alertShowing = alertShowing;
+    }
+
+    public boolean shouldExit() {
+        return shouldExit;
+    }
+
+    public void setShouldExit(boolean shouldExit) {
+        this.shouldExit = shouldExit;
+    }
+
+    public ArrayList<HashMap<String, String>> getFormList() {
+        return formList;
+    }
+
+    public void clearFormList() {
+        formList.clear();
+    }
+
+    public void addFormList(HashMap<String, String> item) {
+        formList.add(item);
+    }
+
+    public void addFormList(int index, HashMap<String, String> item) {
```

Comment:
Let's rename to `addForm` (it's not adding a **new list**, it's adding a **new item** to the existing list)

Reply:
Fixed.